### PR TITLE
512,1024,2048

### PR DESCRIPTION
**Pull request by tmontaigu**

### CODE REVIEW

File: tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs [hunk 1]
```diff
@@ -728,6 +750,9 @@ generic_integer_impl_scalar_operation!(
         (super::FheUint128, u8, u16, u32, u64, u128),
         (super::FheUint160, u8, u16, u32, u64, u128, U256),
         (super::FheUint256, u8, u16, u32, u64, u128, U256),
+        (super::FheUint512, u8, u16, u32, u64, u128, U256, U512),
+        (super::FheUint1024, u8, u16, u32, u64, u128, U256, U512, U1024),
+        (super::FheUint2048, u8, u16, u32, u64, u128, U256, U2048),
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs:755]: likely missing 512 and 1024 for all 2048 calls as it was done before the others were available ? same thing below for all macro calls

File: tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs [hunk 2]
```diff
@@ -1556,6 +1632,9 @@ generic_integer_impl_scalar_operation_assign!(
         (super::FheUint128, u8, u16, u32, u64, u128),
         (super::FheUint160, u8, u16, u32, u64, u128, U256),
         (super::FheUint256, u8, u16, u32, u64, u128, U256),
+        (super::FheUint512, u8, u16, u32, u64, u128, U256, U512),
+        (super::FheUint1024, u8, u16, u32, u64, u128, U256, U1024),
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs:1636]: missing U512

File: tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs [hunk 3]
```diff
@@ -1556,6 +1632,9 @@ generic_integer_impl_scalar_operation_assign!(
...
+        (super::FheUint2048, u8, u16, u32, u64, u128, U256, U2048),
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs:1637]: missing U512, U1024

File: tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs [hunk 4]
```diff
@@ -1591,6 +1670,9 @@ generic_integer_impl_scalar_operation_assign!(
         (super::FheUint128, u8, u16, u32, u64, u128),
         (super::FheUint160, u8, u16, u32, u64, u128, U256),
         (super::FheUint256, u8, u16, u32, u64, u128, U256),
+        (super::FheUint512, u8, u16, u32, u64, u128, U256, U512),
+        (super::FheUint1024, u8, u16, u32, u64, u128, U256, U1024),
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs:1674]: ditto

File: tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs [hunk 5]
```diff
@@ -1626,6 +1708,9 @@ generic_integer_impl_scalar_operation_assign!(
...
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs:1712]: same

File: tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs [hunk 6]
```diff
@@ -1661,6 +1746,9 @@ generic_integer_impl_scalar_operation_assign!(
...
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/unsigned/scalar_ops.rs:1750]: as well

**Review by IceTDrinker — COMMENTED** [review]
Small thing but otherwise will be good to merge

File: tfhe/src/js_on_wasm_api/js_high_level_api/integers.rs [hunk 7]
```diff
@@ -315,14 +315,22 @@ create_wrapper_type_non_native_type!(
...
+    },
     {
         type_name: FheUint2048,
         compressed_type_name: CompressedFheUint2048,
-        compact_type_name: CompactFheUint2048,
-        compact_list_type_name: CompactFheUint2048List,
         proven_type: ProvenFheUint2048,
-        proven_compact_type_name: ProvenCompactFheUint2048,
-        proven_compact_list_type_name: ProvenCompactFheUint2048List,
         rust_type: U2048,
```
[the remark this answers is not in the document]
Comment (tmontaigu) [reply tfhe/src/js_on_wasm_api/js_high_level_api/integers.rs:334]: The first commit is a cherry pick

[the remark this answers is not in the document]
Comment (tmontaigu) [reply tfhe/src/js_on_wasm_api/js_high_level_api/integers.rs]: Merged the commits

**Review by IceTDrinker — APPROVED** [review]
Thanks !

**Pull request merged**